app/code/Magento/Shipping/Model/Carrier/AbstractCarrierInterface.php

```diff
@@ -5,199 +5,12 @@
  */
```

Comment:
Leave methods isStateProvinceRequired, isCityRequired, debugData in this interface. These methods must be also deprecated.

Reply:
As described in this comment by @okorshenko the AbstractCarrierInterface should be empty.
https://github.com/magento/magento2/pull/11293#issuecomment-339484781

Reply:
As described in comment that you mentioned "We need to move all the “correct” methods to CarrierInterface", so I please you to leave unused methods in deprecated interface.